src/components/expenses/Form.tsx

```diff
@@ -0,0 +1,181 @@
+    documentId: data?.documentId || '',
+    approvedById: data?.approvedById || '',
+  }
+
+  const mutationFn = id ? useEditExpense(id) : useCreateExpense()
+
+  const mutation = useMutation<AxiosResponse<ExpenseResponse>, AxiosError<ApiErrors>, ExpenseInput>(
+    {
+      mutationFn,
+      onError: () =>
+        AlertStore.show(id ? t('alerts.edit-row.error') : t('alerts.new-row.error'), 'error'),
+      onSuccess: () => {
+        queryClient.invalidateQueries(endpoints.expenses.listExpenses.url)
+        router.push(routes.admin.expenses.index)
+        AlertStore.show(id ? t('alerts.edit-row.success') : t('alerts.new-row.success'), 'success')
+      },
+    },
+  )
+
+  async function onSubmit(data: ExpenseInput, { setFieldError }: FormikHelpers<ExpenseInput>) {
+    try {
+      if (data.documentId == '') {
+        data.documentId = null
+      }
+      if (data.approvedById == '') {
+        data.approvedById = null
+      }
+      mutation.mutateAsync(data)
+    } catch (error) {
+      if (isAxiosError(error)) {
+        const { response } = error as AxiosError<ApiErrors>
+        response?.data.message.map(({ property, constraints }) => {
+          setFieldError(property, t(matchValidator(constraints)))
+        })
+      }
+    }
+  }
+
+  return (
+    <GenericForm
+      onSubmit={onSubmit}
+      initialValues={initialValues}
+      validationSchema={validationSchema}>
+      <Box sx={{ marginTop: '5%', height: '62.6vh' }}>
+        <Typography variant="h5" component="h2" sx={{ marginBottom: 2, textAlign: 'center' }}>
+          {id ? t('headings.edit') : t('headings.add')}
+        </Typography>
+        <Grid container spacing={2} sx={{ width: 600, margin: '0 auto' }}>
+          <Grid item xs={6}>
+            <ExpenseTypeSelect />
+          </Grid>
+          <Grid item xs={6}>
+            <ExpenseStatusSelect />
+          </Grid>
+          <Grid item xs={6}>
+            <FormTextField type="number" name="amount" label="expenses:fields.amount" />
+          </Grid>
+          <Grid item xs={6}>
+            <CurrencySelect />
+          </Grid>
+          <Grid item xs={6}>
+            <VaultSelect />
+          </Grid>
+          <Grid item xs={6}>
+            <DocumentSelect />
+          </Grid>
+          <Grid item xs={12}>
+            <FormTextField
+              type="string"
+              name="description"
+              label="expenses:fields.description"
+              multiline
+              rows={5}
+            />
+          </Grid>
+          <Grid item xs={id ? 10 : 12}>
+            <PersonSelect name="approvedById" />
+          </Grid>
+          {id ? (
+            <Grid item xs={2}>
+              <DeletedCheckbox />
+            </Grid>
+          ) : null}
+
+          {id ? (
+            <>
+              <Grid item xs={6}>
+                <Button fullWidth variant="contained" type="submit" color="secondary">
+                  {t('btns.save')}
+                </Button>
```

Comment:
Please use  `<SubmitButton fullWidth label="btns.save" />` instead of custom button

Reply:
If the only difference is the label you can simplify this code to:

```tsx
<Grid item xs={6}>
  <SubmitButton fullWidth label={id ? 'btns.save' : 'expenses:btns.submit'} />
</Grid>
```